src/core-services/catalog/mutations/publishProducts.js

```diff
@@ -29,10 +29,8 @@ export default async function publishProducts(context, productIds) {
 
   if (!isInternalCall) {
     const uniqueShopIds = _.uniq(products.map((product) => product.shopId));
-    uniqueShopIds.forEach((shopId) => {
-      if (!userHasPermission(["createProduct", "product/admin", "product/publish"], shopId)) {
-        throw new ReactionError("access-denied", "Access Denied");
-      }
+    uniqueShopIds.forEach(async (shopId) => {
+      await checkPermissions(["createProduct", "product/admin", "product/publish"], shopId);
```

Comment:
Is this OK to do in a `forEach`, or does it need to be done in a `for...of` instead?